lib/execution_engine2/execution_engine2Impl.py

```diff
@@ -41,6 +42,10 @@ class execution_engine2:
 
     ADMIN_ROLES_CACHE_SIZE = 500
     ADMIN_ROLES_CACHE_EXPIRE_TIME = 300  # seconds
+
+    def get_user_clients(self, ctx) -> UserClientSet:
```

Comment:
Is there an advantage of having this here instead of inside the SDKMethodRunner?

Reply:
yes, instantiating the client set outside of SDKMR is what allows for passing mock objects into SDKMR rather than monkey patching (although you can still do that if you want) and enables constructor-based [dependency injection](https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/). The major change for this PR is exactly that - making the code that creates SDKMR responsible for the creation of SDKMR's dependencies rather than SDKMR.

Reply:
I should have said inside the SDKMethodRunner file, defined outside of the SDKMethodRunner class, or moving this function to its own file. We should probably move out everything from the BEGIN_CLASS_HEADER section to its own file, as there are often issues with editing code in the impl file, and we have the common KBase pattern of doing as little as possible in the IMPL file itself.

Reply:
Oh I see. I'll make an APIHelper class that holds a reference to the config so we don't have to pass it around all over place and put the method there.

Reply:
done